File: src/contests/cg/puzzle/python3/the-fastest.py
# 🎮 CodinGame Puzzle - the-fastest
# https://www.codingame.com/training/medium/the-fastest

# Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
n = int(input())
best = ""
for _ in range(n):
    t = input().strip()
    if best == "" or t < best:
        best = t
print(best)

File: src/contests/cg/puzzle/ruby/the-fastest.rb
# 🎮 CodinGame Puzzle - the-fastest
# https://www.codingame.com/training/medium/the-fastest

# Times are HH:MM:SS so lexicographic comparison equals chronological comparison.
n = gets.to_i
best = ""
n.times do
  t = gets.strip
  best = t if best == "" || t < best
end
puts best
